scripts/resolve-includes.groovy

```diff
@@ -157,9 +160,50 @@ private void copyInvokerPropertiesToProjects(List<String> projects) {
     }
 }
 
+private List<String> resolveProjectExplicitFiltering(ArrayList<String> leafProjects) {
+    List<String> projects = mavenProjectExplicitFiltering.split(',')
+    List<String> filteredLeafProjects = new ArrayList<>()
+    List<String> filteredOutLeafProjects = new ArrayList<>()
+    String filteringTYpe = mavenProjectExplicitFilteringType
+
+    def logFile = "filter-projects.log"
+    log.info("Starting explicit filtering projects resolution, results will be in ${Path.of(logdir).resolve(logFile).toString()}")
+
+    if (projects && !projects.empty) {
+        for (int i = 0; i < leafProjects.size(); i++) {
+            String current = leafProjects[i]
+            for (int j = 0; j < projects.size(); j++) {
+                String filter = projects[j]
+
+                if (Path.of(current).endsWith(filter)) {
+                    if (filteringTYpe.equals("in")) {
+                        filteredLeafProjects.add(current)
+                        log.debug("Adding $current as a selected project for tests")
+                    } else if (filteringTYpe.equals("out")) {
+                        filteredOutLeafProjects.add(current)
+                        log.debug("Skipping '$current' project, cause it has been filtered out")
+                    }
+                }
+            }
+        }
+    }
+
+    if (filteringTYpe.equals("in")) {
```

Comment:
Considers only "in" or "out" values. Returns the selected modules when "in" is selected. Returns all modules but the filtered out modules when "out" is selected. Or returns the standard list when nothing ("in" or "out") is selected.